pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -100,6 +102,8 @@ public ServerCnx(BrokerService service) {
         this.producers = new ConcurrentLongHashMap<>(8, 1);
         this.consumers = new ConcurrentLongHashMap<>(8, 1);
         this.replicatorPrefix = service.pulsar().getConfiguration().getReplicatorPrefix();
+        this.nonPersistentMessageSemaphore = new Semaphore(
```

Comment:
Actually, for non-persistent topic it is been called from [ordered-executor thread](https://github.com/rdhabalia/pulsar/blob/50fdd31dd78b1fbe9bffd9d083108903bfe82d97/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java?utf8=%E2%9C%93#L211). So, counter will be incremented by io-thread and decrement by ordered-thread. Not, sure if it would be thread-safe in terms throttling with correct count. 
so, do you still think it would be fine to keep one unlock counter variable would be fine here or it should be atomic?